app/models/submission.rb

```diff
@@ -248,6 +250,20 @@ def new_version_needed?
       )
     }
     can :view_turnitin_report
+
+    given { |user, session|
+      self.vericiteable? && vericite_data_hash &&
+      user_can_read_grade?(user, session) &&
+      (assignment.context.grants_right?(user, session, :manage_grades) ||
+        case assignment.vericite_settings[:originality_report_visibility]
+          when 'immediate'; true
+          when 'after_grading'; current_submission_graded?
+          when 'after_due_date'; assignment.due_at && assignment.due_at < Time.now.utc
+          when 'never'; false
+        end
+      )
+    }
+    can :view_vericite_report
```

Comment:
This seems like it should be generic.  Maybe :view_plagiarism_report with combined logic and/or helper functions.